comms/core/src/connectivity/proactive_dialer.rs

```diff
@@ -74,24 +74,29 @@ impl ProactiveDialer {
             return Ok(0);
         }
 
-        let current_connections = pool.count_connected_nodes();
+        let eligible_connections = pool
+            .get_connected_node_ids()
+            .iter()
+            .filter(|p| !excluded_peers.contains(p))
+            .count();
         let target = self.config.target_connection_count;
 
         // Update metrics
 
-        if current_connections >= target {
+        if eligible_connections >= target {
```

Comment:
this is wrong. 
Should stay as it was, 
we need x connections, if some are seeds, why might have a different reason why we are connected to them, but thats fine and when we disconnect we will find another connection. We dont wants X + seeds connections as the connections